src/FSharpPlus/Data/Validation.fs

```diff
@@ -243,6 +243,29 @@ module Validation =
         List.iter (function Success e -> coll1.Add e | Failure e -> coll2.Add e) source
         coll1.Close (), coll2.Close ()
     #endif
+    
+    [<AutoOpen>]
+    module ComputationExpression =
+        type ValidationBuilder() =
+            member _.Bind(x, f) =
+                match x with
+                | Failure e -> Failure e
+                | Success a -> f a
+
+            member _.MergeSources(left : Validation<list<string>, _>, right : Validation<list<string>, _>) =
+                match left, right with
+                | Success l, Success r -> Success (l, r)
+                | Failure l, Success _ -> Failure l
+                | Success _, Failure r -> Failure r
+                | Failure r, Failure l -> List.append r l |> Failure
+
+            member _.Return(x) =
+                Success x
+                
+            member _.ReturnFrom(x) =
+                x
+
+        let validator = ValidationBuilder()
```

Comment:
Note that this is just a special case of the `applicative` computation expression, closed to `Validation<list<string>, 'T>`.

I mean, right now you can just define it like: `let validator<'T> = applicative<Validation<list<string>, 'T>>` and you'll have the same methods and more available.

Reply:
Actually the `bind` is not part of the `applicative` CE, but `Validation` is not a monad, as it doesn't obey monad rules. However in order to get the short-circuit behavior you can use `let validator<'T> = monad'<Result<'T, List<string>>`

Reply:
Maybe an alias then?